config/config.go

```diff
@@ -134,7 +134,7 @@ func NewConfig() (c *Config) {
 	c.Server.CORSAllowAll = getDefaultBool(EnvServerCORSAllowAll, false)
 
 	// Prometheus configuration
-	c.ExternalServices.PrometheusServiceURL = strings.TrimSpace(getDefaultString(EnvPrometheusServiceURL, "http://prometheus:9090"))
+	c.ExternalServices.PrometheusServiceURL = strings.TrimSpace(getDefaultString(EnvPrometheusServiceURL, "http://prometheus.istio-system:9090"))
```

Comment:
I guess this is a change in OS 3.9 ? right ? as in OS 3.7 the prometheus was resolved within the same namespace.

Reply:
Not really, it could have been done since the beginning. Both urls are still correct, but giving the namespace makes it agnostic to kiali's own namespace when connecting to prometheus. That will help when we'll want to consider if kiali can be deployed in another namespace

(actually, with this change it seems to be working pretty well when deployed elsewhere, I haven't checked extensively if something else is broken but most features seem ok)

In any case this change doesn't hurt imo.